pipeline/api/asyncio/cloud.py

```diff
@@ -0,0 +1,220 @@
+
+        return response.json()
+
+    async def run_pipeline(
+        self,
+        pipeline_id_or_schema: Union[str, PipelineGet],
+        raw_data_or_schema: Union[Any, DataGet],
+        compute_type: Optional[str] = None,
+        min_gpu_vram_mb: Optional[int] = None,
+    ):
+        """
+        Uploads Data and executes a Run of given pipeline over given data.
+
+            Parameters:
+                    pipeline_id_or_schema (Union[str, PipelineGet]):
+                        The id for the desired pipeline
+                        or the schema obtained from uploading it
+                    raw_data_or_schema (Union[Any, DataGet]):
+                        Raw data for Pipeline execution
+                        or schema for already uploaded data.
+                    compute_type (Optional[str]):
+                        Compute type requirement for this run (e.g. "cpu" or "gpu"). If
+                        not defined, it will fallback to that specified by the Pipeline
+                        itself (typically this is "gpu")
+                    min_gpu_vram_mb (Optional[int]):
+                        Minimum amount of GPU VRAM required. If not defined, it will
+                        fallback to that specified by the Pipeline (if applicable). This
+                        should only be specified for GPU workloads.
+
+            Returns:
+                    run (Any): Run object containing metadata and outputs.
+        """
+        # TODO: Add support for generic object inference. Only strs at the moment.
+        if not isinstance(raw_data_or_schema, DataGet):
+            temp_file = io.BytesIO(python_object_to_hex(raw_data_or_schema).encode())
+            uploaded_data = await self.upload_data(temp_file, "/")
+            _data_id = uploaded_data.id
+        elif isinstance(raw_data_or_schema, DataGet):
+            _data_id = raw_data_or_schema.id
+        else:
+            raise Exception("Must either pass a raw data, or DataGet schema.")
+
+        pipeline_id = None
+        if isinstance(pipeline_id_or_schema, str):
+            pipeline_id = pipeline_id_or_schema
+        elif isinstance(pipeline_id_or_schema, PipelineGet):
+            pipeline_id = pipeline_id_or_schema.id
+        else:
+            raise InvalidSchema(
+                schema=pipeline_id_or_schema,
+                message=(
+                    "Must either pass a pipeline id, or a pipeline get schema. "
+                    "Not object of type %s in arg 1." % str(pipeline_id_or_schema)
+                ),
+            )
+
+        compute_requirements = None
+        if min_gpu_vram_mb:
+            compute_requirements = ComputeRequirements(min_gpu_vram_mb=min_gpu_vram_mb)
+
+        run_create_schema = RunCreate(
+            pipeline_id=pipeline_id,
+            data_id=_data_id,
+            compute_type=compute_type,
+            compute_requirements=compute_requirements,
+        )
```

Comment:
I think it would still be good to reduce some of this code duplication but we can do that in a follow up PR